contracts/InterchainTokenService.sol

```diff
@@ -731,15 +751,15 @@ contract InterchainTokenService is
     /**
      * @notice Processes a deploy token manager payload.
      */
-    function _processDeployTokenManagerPayload(bytes memory payload) internal {
-        (, bytes32 tokenId, TokenManagerType tokenManagerType, bytes memory params) = abi.decode(
+    function _processLinkTokenPayload(bytes memory payload) internal {
+        (, bytes32 tokenId, TokenManagerType tokenManagerType, , bytes memory destinationTokenAddress, , bytes memory linkParams) = abi.decode(
             payload,
-            (uint256, bytes32, TokenManagerType, bytes)
+            (uint256, bytes32, TokenManagerType, bytes, bytes, bool, bytes)
         );
 
         if (tokenManagerType == TokenManagerType.NATIVE_INTERCHAIN_TOKEN) revert CannotDeploy(tokenManagerType);
 
-        _deployTokenManager(tokenId, tokenManagerType, params);
+        _deployTokenManager(tokenId, tokenManagerType, destinationTokenAddress.toAddress(), linkParams);
```

Comment:
I think even if the only param left in EVM is the operator, we should have it be abi encoded when it is passed as params.